src/js/components/MultiSelect/MultiSelect.js

```diff
@@ -0,0 +1,504 @@
+      () =>
+        getDisplayLabelKey(
+          labelKey,
+          allOptions,
+          optionIndexesInValue,
+          selectValue,
+        ),
+      [labelKey, allOptions, optionIndexesInValue, selectValue],
+    );
+
+    // text to show
+    // When the options array contains objects, this property indicates how
+    // to retrieve the value of each option.
+    // If a string is provided, it is used as the key to retrieve a
+    // property of an option object.
+    // If a function is provided, it is called with the option and should
+    // return the value.
+    // If reduce is true, this value will be used for the 'value'
+    // delivered via 'onChange'.
+    const inputValue = useMemo(() => {
+      if (!selectValue) {
+        if (optionIndexesInValue.length === 0) return '';
+        if (optionIndexesInValue.length === 1)
+          return applyKey(allOptions[optionIndexesInValue[0]], labelKey);
+        if (messages) return format({ id: 'select.multiple', messages });
+        return `${optionIndexesInValue.length} selected`;
+      }
+      return undefined;
+    }, [
+      labelKey,
+      messages,
+      format,
+      optionIndexesInValue,
+      allOptions,
+      selectValue,
+    ]);
+
+    const iconColor = getIconColor(theme);
+
+    const dropContent = (
+      <MultiSelectContainer
+        allOptions={allOptions}
+        disabled={disabled}
+        disabledKey={disabledKey}
+        dropHeight={dropHeight}
+        emptySearchMessage={emptySearchMessage}
+        help={help}
+        id={id}
+        labelKey={labelKey}
+        limit={limit}
+        onChange={onSelectChange}
+        onClose={onRequestClose}
+        onKeyDown={onKeyDown}
+        onMore={onMore}
+        onSearch={onSearch}
+        options={optionsProp}
+        optionIndexesInValue={optionIndexesInValue}
+        replace={replace}
+        searchPlaceholder={searchPlaceholder}
+        search={search}
+        setSearch={setSearch}
+        usingKeyboard={usingKeyboard}
+        value={value}
+        valueKey={valueKey}
+        showSelectedInline={showSelectedInline}
+      >
+        {children}
+      </MultiSelectContainer>
+    );
+
+    const dropButtonProps = {
+      ref: dropButtonRef,
+      a11yTitle: `${ariaLabel || a11yTitle || placeholder || 'Open Drop'}. ${
+        value.length
+      } selected.`,
+      'aria-expanded': Boolean(open),
+      'aria-haspopup': 'listbox',
+      id,
+      disabled: disabled === true || undefined,
+      open,
+      focusIndicator,
+      onFocus,
+      onBlur,
+      gridArea,
+      margin,
+      onOpen: onRequestOpen,
+      onClose: onRequestClose,
+      onClick,
+      callerPlain: plain,
+      plain, // Button should be plain
+      dropProps,
+      dropContent,
+      theme,
+    };
+
+    const displaySelectIcon = SelectIcon && (
+      <Box
+        alignSelf="center"
+        margin={theme.select.icons.margin}
+        style={{ minWidth: 'auto' }}
```

Comment:
we have a minWidth prop on `Box` did that not work the same?
`width={{ min: 'auto' }}`

Reply:
not sure if `auto` is an option in grommet though

Reply:
`width={{ min: 'auto' }}` does work, I wasn't thinking when I made the inline style. I switched it in latest push